pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -196,7 +199,7 @@ def transfer_distrib_spawn_state_on_fit_end(self, results):
                 and len(best_model_path) > 0
             ):
                 last_path = re.sub(".ckpt", ".tmp_end.ckpt", best_model_path)
-                self.save(self.lightning_module.state_dict(), last_path)
+                self.save(state_dict, last_path)
```

Comment:
Changes were not required for `tpu_spawn`. As `self.save` is called for all ranks, but this looks good and clean as well.

Reply:
Okay, thanks. Then you recommend to still keep it like this?

Reply:
Yup, the change is fine.